packages/core/admin/admin/src/hooks/useEnterprise/useEnterprise.js

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+
+import { useCallbackRef } from '@strapi/helper-plugin';
+
+function isEnterprise() {
+  return window.strapi.isEE;
+}
+
+export function useEnterprise(
+  ceData,
+  eeCallback,
+  { defaultValue = null, combine = (ceData, eeData) => eeData } = {}
+) {
+  const eeCallbackRef = useCallbackRef(eeCallback);
+  const combineCallbackRef = useCallbackRef(combine);
+
+  // We have to use a nested object here, because functions (e.g. Components)
+  // can not be stored as value directly
+  const [{ data }, setData] = React.useState({
+    data: isEnterprise() ? defaultValue : ceData,
+  });
+
+  React.useEffect(() => {
+    async function importEE() {
+      const eeData = await eeCallbackRef();
+
+      if (Object.keys(eeData).length > 1) {
+        throw new Error('Imported EE module has more than one export.');
+      } else if (Object.keys(eeData).length === 0) {
+        throw new Error('Imported EE module does not have an export.');
+      }
```

Comment:
As we discussed briefly this morning, I think we shouldn't be looking for all this, your `eeCallback` should define what you return to be more inline with how Dynamic Imports work in the spec, we could throw if you don't give anything? This would mean named exports should work ootb.